pkg/broker/broker.go

```diff
@@ -759,23 +742,20 @@ func (a AnsibleBroker) Bind(instanceUUID uuid.UUID, bindingUUID uuid.UUID, req *
 	}
 	// Can't bind to anything if we have nothing to return to the catalog
 	if provExtCreds == nil && bindExtCreds == nil {
-		a.log.Error("No extracted credentials found from provision or bind")
-		a.log.Error("Instance ID: %s", instanceUUID.String())
+		a.log.Errorf("No extracted credentials found from provision or bind instance ID: %s",
+			instanceUUID.String())
 		return nil, errors.New("No credentials available")
 	}
 
-	returnCreds := mergeCredentials(provExtCreds, bindExtCreds)
-	// TODO: Insert merged credentials into etcd? Separate into bind/provision
-	// so none are overwritten?
-
-	return &BindResponse{Credentials: returnCreds}, nil
-}
-
-func mergeCredentials(provExtCreds *apb.ExtractedCredentials,
-	bindExtCreds *apb.ExtractedCredentials,
-) map[string]interface{} {
-	// TODO: Implement, need to handle case where either are empty
-	return provExtCreds.Credentials
+	if bindExtCreds != nil {
+		err = a.dao.SetExtractedCredentials(bindingUUID.String(), bindExtCreds)
+		if err != nil {
+			a.log.Errorf("Could not persist extracted credentials - %v", err)
+			return nil, err
+		}
+		return &BindResponse{Credentials: bindExtCreds.Credentials}, nil
```

Comment:
This is better than our original thinking. Keeps the broker dumb and delegates the responsibility for what needs to be in the bind response entirely to the APBs, while still allowing them to include provision credentials since they're passed in as a param during the bind action. Nice job :+1: